docs/source/customizations/rhai-api.mdx

```diff
@@ -0,0 +1,328 @@
+---
+title: Rhai script API reference
```

Comment:
I broke the most reference-y portions of the article out into a _separate_ article to hopefully ease the onboarding in the conceptual doc